src/main/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStep.java

```diff
@@ -101,7 +101,7 @@ public final void checkout(Run<?,?> run, FilePath workspace, TaskListener listen
             if (changelog) {
                 for (int i = 0; ; i++) {
                     changelogFile = new File(run.getRootDir(), "changelog" + i + ".xml");
-                    if (!changelogFile.exists()) {
+                    if (changelogFile.createNewFile()) {
```

Comment:
As I noted in https://github.com/jenkinsci/workflow-scm-step-plugin/pull/32#discussion_r249973975, this seems like it could introduce its own problems.

Reply:
I see, I didn't take in consideration failed checkouts, nor that empty changelog files posed a threat. I can go with #32, fixes the same issue, but the proposed change needs some tweaking. see https://github.com/jenkinsci/workflow-scm-step-plugin/pull/32#issuecomment-465129978